File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stk = new Stack<>();
        boolean answer = true;
        int len = s.length();
        for (int i = 0; i < len; i++) {
            char curr = s.charAt(i);
            if (curr == '(') {
                stk.push(curr);
            }
            
            if (curr == ')' && !stk.isEmpty()) {
                stk.pop();
            } else if (curr == ')' && stk.isEmpty()) {
                answer = false;
                break;
            } 
        }
        if (!stk.isEmpty()) answer = false;
        return answer;

    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    
    const arr = [];
    
    var answer = true;

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    for (i = 0; i < s.length; i++) {
        let cur = s.charAt(i);
        if (cur === ')') {
            if (arr[arr.length - 1] === '(') {
                arr.pop();
            }
            else {
                return false;
            }
        }
        if (cur === '(') {
            arr.push(cur);
        }
    }
    if (arr.length > 0) return false;
    
    return answer;
}